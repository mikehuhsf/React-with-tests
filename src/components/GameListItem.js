import React, { useState } from 'react';
import PropTypes from "prop-types";
import DoctorDetails from "./GameDetails"
import GameDetails from './GameDetails';

function GameListItem({id, title}) {
    const [details, setDetails] =useState(null); 

    function handleLoadDetails() {
        //fetch(`http://localhost:5000/v1/doctor/${id}`)
        fetch(`https://stormy-eyrie-59187.herokuapp.com/v1/doctor/${id}`)
          .then((response) => response.json())
          .then((response) => setDetails(response));
      }

    return (
    <div>
        <a href="#" onClick={handleLoadDetails}>
            {title}
        </a>
        {
            details && (
                <GameDetails
                release_date = {details.release_date}
                platforms = {details.platforms}
                />
            )
        }
    </div>
    );
}

GameListItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
}

export default GameListItem