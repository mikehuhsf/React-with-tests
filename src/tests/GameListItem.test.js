import React from 'react';
import renderer from 'react-test-renderer';
import GameListItem from "../components/GameListItem";
import { shallow } from '../enzyme';

it("renders a drop down when clicked", () => {
    const wrapper = renderer.create(<GameListItem id={1} title='amongus'/>);
    expect(wrapper).toMatchSnapshot();
})

it('should find 1 button named the input title', () => {
    const wrapper = shallow(<GameListItem id={1} title='amongus' />);
    //console.log(wrapper);
    expect(wrapper.find('.game-button').render().text()).toEqual('amongus');
  });
  
//   it('should drop down details after a click', () => {
//     const wrapper = shallow(<GameListItem id={1} title='Minecraft' />);
//     //console.log(wrapper);
//     wrapper.find('.game-button').simulate('click');
//     wrapper.update();
//     renderer.create(wrapper);
//     expect(wrapper.find('.details').render().text()).toEqual('Current count: 2');
//   });