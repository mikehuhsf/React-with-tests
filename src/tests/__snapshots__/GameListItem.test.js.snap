// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a drop down when clicked 1`] = `
<div>
  <button
    className="game-button"
    onClick={[Function]}
  >
    amongus
  </button>
</div>
`;
